sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -633,16 +633,16 @@ case class ShowCreateTableCommand(table: TableIdentifier) extends RunnableComman
   }
 
   private def showCreateHiveTable(metadata: CatalogTable): String = {
-    def reportUnsupportedError(): Unit = {
-      throw new UnsupportedOperationException(
+    def reportUnsupportedError(features: Seq[String]): Unit = {
+      throw new AnalysisException(
         s"Failed to execute SHOW CREATE TABLE against table ${metadata.identifier.quotedString}, " +
-          "because it contains table structure(s) (e.g. skewed columns) that Spark SQL doesn't " +
-          "support yet."
+          "which is created by Hive and uses the following feature(s) that are not yet supported " +
+          "by Spark SQL:\n" + features.map(" - " + _).mkString("\n")
```

Comment:
It sounds like that we are missing functionalities. Can we update the error message?